src/main/java/build/buildfarm/worker/shard/CFCExecFileSystem.java

```diff
@@ -308,14 +315,63 @@ private static void checkExecErrors(Path path, List<Throwable> errors) throws Ex
     }
   }
 
+  // binary search of our sorted list of dirs
```

Comment:
we could use `Collections.binarySearch()` or something equivalent

Reply:
Yup. Working with that method is always awkward, but Lists.transform from guava might actually make it palatable (confers RandomAccess).